lib/ReactViews/DataCatalog/DataCatalogItem.jsx

```diff
@@ -76,12 +75,12 @@ export const DataCatalogItem = observer(
             this.props.terria.workbench.contains(this.props.item) &&
             !keepCatalogOpen
           ) {
-            logDatasetAnalyticsEvent(
-              this.props.terria,
-              this.props.item,
-              toAdd ? "addFromCatalogue" : "removeFromCatalogue"
-            );
             this.props.viewState.closeCatalog();
+            this.props.terria.analytics?.logEvent(
```

Comment:
Come to think of it, isn't analytics **always** available on terria?

Reply:
(.. except I now realise terria isn't marked as a required prop in its prop types)

Reply:
Fix this here:https://github.com/TerriaJS/terriajs/issues/4122